src/js/form-builder.js

```diff
@@ -1161,6 +1165,9 @@
       $clone.attr('name', cloneName);
       $clone.addClass('cloned');
       $('.sortable-options', $clone).sortable();
+    //+gimigliano
+      if (opts.typeUserEvents[type] && opts.typeUserEvents[type]['onclone']) opts.typeUserEvents[type]['onclone']($('#'+lastID));

```

Comment:
I did make a couple small changes after merging.

1. `type` was undefined here so I defined it above
2. `onclone` and `onadd` now receive the JS DOM object instead of performing jQuery lookup on ID. There may uses where developer needs only value or something, and passing existing reference is slightly faster.

Everything else is the same. This is very useful. :+1: